Firestore/Source/Model/FSTFieldValue.h

```diff
@@ -194,15 +198,15 @@ typedef NS_ENUM(NSInteger, FSTServerTimestampBehavior) {
  */
 @interface FSTGeoPointValue : FSTFieldValue <FIRGeoPoint *>
 + (instancetype)geoPointValue:(FIRGeoPoint *)value;
-- (FIRGeoPoint *)valueWithOptions:(FSTFieldValueOptions *)options;
+- (FIRGeoPoint *)value;
```

Comment:
So I'm having trouble figuring out why we were declaring valueWithOptions before and why you've changed it to declare value now...  Both are declared on the superclass (FSTFieldValue) as returning T and so there should be no need to declare them here, right?  Or am I confused?

FWIW (no action required)- in JS / Android FieldValue is not generic (I was surprised to see that it is in iOS), in which case it makes a little bit of sense for each subclass to declare the `value` methods so that they can refine the return type... except per my previous comments I think ultimately we should only be using `value` / `valueWithOptions` to convert to user-facing types and therefore we shouldn't need a return type more specific than `id`...

Reply:
I removed these declarations, and the test suite still runs fine. These might be leftovers: it looks like generics were added pretty recently in [this commit](https://github.com/firebase/firebase-ios-sdk/commit/64066c643d4e1fff7d00d656201d056f10fb2c36), and some of these `valueWithOptions` were removed, but some were kept. My guess is that all of them should have been removed.